falcon/asgi/response.py

```diff
@@ -415,8 +415,13 @@ def _asgi_headers(self, media_type=None):
             headers['content-type'] = media_type
 
         try:
-            items = [(n.encode('ascii'), v.encode('ascii')) for n, v in headers.items()]
+            # NOTE(vytas): Supporting ISO-8859-1 for historical reasons as per
+            #   RFC 7230, Section 3.2.4; and to strive for maximum
+            #   compatibility with WSGI.
+            items = _encode_items_to_latin1(headers)
```

Comment:
I assume the overhead of the function call itself is eclipsed by the gain in performance with Cython?

Reply:
Yes, and the gain is more than threefold. I was profiling this carefully when developing, but still, I wasn't referencing the final function from cythonized Falcon.

Adding a targeted benchmark here which does just that:
```python
import timeit

from falcon.util.misc import _encode_items_to_latin1


TEST = {
    'content-length': '4',
    'content-type': 'text/plain',
    'x-another': 'Link_aa_bb_x=1',
}

print([(n.encode('ascii'), v.encode('ascii')) for n, v in TEST.items()])
print(_encode_items_to_latin1(TEST))

print(timeit.repeat(
    lambda: [(n.encode('ascii'), v.encode('ascii')) for n, v in TEST.items()]))
print(timeit.repeat(
    lambda: _encode_items_to_latin1(TEST)))
print(timeit.repeat(
    lambda: [(n.encode('ascii'), v.encode('ascii')) for n, v in TEST.items()]))
print(timeit.repeat(
    lambda: _encode_items_to_latin1(TEST)))
```

```python
[(b'content-length', b'4'), (b'content-type', b'text/plain'), (b'x-another', b'Link_aa_bb_x=1')]
[(b'content-length', b'4'), (b'content-type', b'text/plain'), (b'x-another', b'Link_aa_bb_x=1')]
[0.7468254599999682, 0.7431183099999998, 0.7475773680000657, 0.7449554410000019, 0.7483860550000827]
[0.22064724300003036, 0.2204902519999905, 0.22048896700016485, 0.22051541400014685, 0.2205529090001619]
[0.7458620350000729, 0.7445737529999406, 0.7403353429999697, 0.7462570249999771, 0.7480042119998416]
[0.22065620099988337, 0.22055909699997756, 0.22055948600018382, 0.22050846599995566, 0.22057072500001595]
```

Reply:
I guess the gap increases the more elements are in the db

Reply:
Very nice.